src/visualize/image_like.jl

```diff
@@ -127,35 +109,114 @@ function play{T}(buffer::Array{T, 2}, video_stream, t)
     return reinterpret(T, buffer, (w,h))
 end
 
-"""
-Takes a 3D image and decides if it is a volume or an animated Image.
-"""
-function _default{T<:Colorant, X}(img::Images.Image{T, 3, X}, s::Style, data::Dict)
-    props = img.properties
-    if haskey(props, "timedim")
-        if haskey(props, "spatialorder")
-            get!(data, :spatialorder, join(props["spatialorder"], ""))
+# If the user is using the new Images, ImageAxes will be loaded
+if isdefined(Images, :ImageAxes)
+    unwrap(img::Images.ImageMeta) = unwrap(data(img))
+    unwrap(img::AxisArray) = unwrap(img.data)
+    unwrap(img::AbstractArray) = img
+
+    GLAbstraction.gl_convert{T}(::Type{T}, img::AbstractArray) = _gl_convert(T, unwrap(img))
+    _gl_convert{T}(::Type{T}, img::Array) = gl_convert(T, img)
+
+    """
+        play(img)
+
+    Turns an Image into a video stream
+    """
+    function play{T}(img::HasAxesArray{T, 3})
+        ax = timeaxis(img)
+        if timeaxis(img) != nothing
+            return const_lift(play, unwrap(img), timedim(img), loop(1:length(ax)))
         end
-        timedim = props["timedim"]
-        video_signal = const_lift(play, img.data, timedim, loop(1:size(img, timedim)))
-        return _default(video_signal, s, data)
-    elseif haskey(props, "pixelspacing")
-        spacing = Vec3f0(map(float, img.properties["pixelspacing"]))
-        pdims   = Vec3f0(size(img))
+        error("Image has no time axis: axes(img) = $(axes(img))")
+    end
+
+    """
+    Takes a 3D image and decides if it is a volume or an animated Image.
+    """
+    function _default{T}(img::HasAxesArray{T,3}, s::Style, data::Dict)
+        # We could do this as a @traitfn, except that those don't
+        # currently mix well with non-trait specialization.
+        if timeaxis(img) != nothing
+            data[:spatialorder] = "yx"
+            timedim = timedim(img)
+            video_signal = const_lift(play, unwrap(img), timedim, loop(1:size(img, timedim)))
+            return _default(video_signal, s, data)
+        else
+            ps = pixelspacing(img)
+            spacing = Vec3f0(map(x->x/maximum(ps), ps))
+            pdims   = Vec3f0(map(length, indices(img)))
+            dims    = pdims .* spacing
+            dims    = dims/maximum(dims)
+            data[:dimensions] = dims
+            _default(unwrap(img), s, data)
+        end
+    end
+
+    function _default{T}(img::HasAxesArray{T,2}, s::Style, data::Dict)
```

Comment:
This is intended to allow even 2d images to use pixel spacing that is not 1:1. This will be important for accurate rendering of slices in the upcoming "volumes.jl" demo, where the pixelspacing in 3d is (1,1,5) and where two of the 2d slices look wrong without fixing the geometry. However, this is not actually working to solve that problem. Any tips?